src/pip/_internal/locations/_sysconfig.py

```diff
@@ -143,10 +143,13 @@ def get_scheme(
         python_xy = f"Python{sys.version_info.major}{sys.version_info.minor}"
         paths["scripts"] = os.path.join(base, python_xy, "Scripts")
 
+    # Substitute an empty name with "UNKNOWN" for distutils compatibility.
+    headers = os.path.join(paths["include"], dist_name or "UNKNOWN")
```

Comment:
Wait, what?

I thought the new behaviour was "more correct", and it'd be worthwhile to be a little disruptive about it.

Reply:
Strictly speaking, there isn’t really a “more correct” behaviour here. This is trying to get a header path for a distribution, and a distribution without a name shouldn’t have a header path in the first place.

In practice, this case is hit when pip tries to get some other paths (e.g. to get the `scripts` directory) and doesn’t really care about headers. Those calls should ideally be changed to use something that makes more sense instead, but there are *a lot* of them.

Reply:
The warning message about this is not just “a little” disruptive. It’s pretty likely to be shown much too often things and flood over actual issue reports.